src/measure_theory/measure/with_density_vector_measure.lean

```diff
@@ -0,0 +1,140 @@
+  end }
+else 0
+
+open measure
+
+include m
+
+variables {f g : α → E}
+
+lemma with_densityᵥ_apply (hf : integrable f μ) {s : set α} (hs : measurable_set s) :
+  μ.with_densityᵥ f s = ∫ x in s, f x ∂μ :=
+by { rw [with_densityᵥ, dif_pos hf], exact dif_pos hs }
+
+@[simp]
+lemma with_densityᵥ_zero : μ.with_densityᵥ (0 : α → E) = 0 :=
+begin
+  ext1 s hs,
+  erw [with_densityᵥ_apply (integrable_zero α E μ) hs],
+  simp,
+end
+
+@[simp]
+lemma with_densityᵥ_neg : μ.with_densityᵥ (-f) = -μ.with_densityᵥ f :=
+begin
+  by_cases hf : integrable f μ,
+  { ext1 i hi,
+    rw [vector_measure.neg_apply, with_densityᵥ_apply hf hi,
+        ← integral_neg, with_densityᵥ_apply hf.neg hi],
+    refl },
+  { rw [with_densityᵥ, with_densityᵥ, dif_neg hf, dif_neg, neg_zero],
+    rwa integrable_neg_iff }
+end
+
+@[simp]
+lemma with_densityᵥ_add (hf : integrable f μ) (hg : integrable g μ) :
+  μ.with_densityᵥ (f + g) = μ.with_densityᵥ f + μ.with_densityᵥ g :=
+begin
+  ext1 i hi,
+  rw [with_densityᵥ_apply (hf.add hg) hi, vector_measure.add_apply,
+      with_densityᵥ_apply hf hi, with_densityᵥ_apply hg hi],
+  simp_rw [pi.add_apply],
+  rw integral_add; rw ← integrable_on_univ,
+  { exact hf.integrable_on.restrict measurable_set.univ },
+  { exact hg.integrable_on.restrict measurable_set.univ }
+end
+
+@[simp]
+lemma with_densityᵥ_sub (hf : integrable f μ) (hg : integrable g μ) :
+  μ.with_densityᵥ (f - g) = μ.with_densityᵥ f - μ.with_densityᵥ g :=
+by rw [sub_eq_add_neg, sub_eq_add_neg, with_densityᵥ_add hf hg.neg,
+       with_densityᵥ_neg]
+
+@[simp]
+lemma with_densityᵥ_smul {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+  [normed_space 𝕜 E] [smul_comm_class ℝ 𝕜 E] [measurable_space 𝕜] [opens_measurable_space 𝕜]
+  (r : 𝕜) : μ.with_densityᵥ (r • f) = r • μ.with_densityᵥ f :=
+begin
+  by_cases hf : integrable f μ,
+  { ext1 i hi,
+    rw [with_densityᵥ_apply (hf.smul r) hi, vector_measure.smul_apply,
+        with_densityᵥ_apply hf hi, ← integral_smul r f],
+    refl },
+  { by_cases hr : r = 0,
+    { subst hr, rw [zero_smul, zero_smul, with_densityᵥ_zero] },
+    { rw [with_densityᵥ, with_densityᵥ, dif_neg hf,
+          dif_neg, smul_zero],
+      rwa integrable_smul_iff hr f } }
+end
+
+lemma measure.with_densityᵥ_absolutely_continuous (μ : measure α) (f : α → ℝ) :
+  μ.with_densityᵥ f ≪ μ.to_ennreal_vector_measure :=
+begin
+  by_cases hf : integrable f μ,
+  { refine vector_measure.absolutely_continuous.mk (λ i hi₁ hi₂, _),
+    rw to_ennreal_vector_measure_apply_measurable hi₁ at hi₂,
+    rw [with_densityᵥ_apply hf hi₁, measure.restrict_zero_set hi₂,
+        integral_zero_measure] },
+  { rw [with_densityᵥ, dif_neg hf],
+    exact vector_measure.absolutely_continuous.zero _ }
+end
+
+/-- Having the same density implies the underlying functions are equal almost everywhere. -/
+lemma integrable.ae_eq_of_with_densityᵥ_eq {f g : α → E}
+  (hf : integrable f μ) (hg : integrable g μ)
+  (hfg : μ.with_densityᵥ f = μ.with_densityᵥ g) : f =ᵐ[μ] g :=
```

Comment:
```suggestion
lemma integrable.ae_eq_of_with_densityᵥ_eq {f g : α → E} (hf : integrable f μ) (hg : integrable g μ)
  (hfg : μ.with_densityᵥ f = μ.with_densityᵥ g) :
  f =ᵐ[μ] g :=
```

Reply:
You could also provide an iff version of that lemma.

Reply:
Done!